tests/foreman/maintain/test_advanced.py

```diff
@@ -0,0 +1,434 @@
+            f'hammer -u admin -p admin user update --login admin --password {default_admin_pass}'
+        )
+        assert result.status == 0
+        # Make default admin creds available in MAINTAIN_HAMMER_YML
+        result = sat_maintain.cli.Advanced.run_hammer_setup()
+        assert result.status == 0
+        # Make sure default password available in MAINTAIN_HAMMER_YML
+        result = sat_maintain.execute(
+            f"grep -i ':password: {default_admin_pass}' {MAINTAIN_HAMMER_YML}"
+        )
+        assert result.status == 0
+        assert default_admin_pass in result.stdout
+
+
+@pytest.mark.stubbed
+def test_positive_satellite_maintain_packages_update(sat_maintain):
+    """Packages update using advanced procedure run
+
+    :id: d56523d7-7042-40e1-a96e-db8f33b960e5
+
+    :steps:
+        1. Run satellite-maintain advanced procedure run packages-update
+
+    :expectedresults: packages should update.
+    """
+    result = sat_maintain.cli.Advanced.run_packages_update()
+    assert 'FAIL' not in result.stdout
+
+
+def test_positive_foreman_taks_delete_old(sat_maintain):
+    """Delete old foreman-tasks using advanced procedure run
+
+    :id: c5deaf67-8fa9-43a6-bf62-1c0d6be63a85
+
+    :steps:
+        1. Run satellite-maintain advanced procedure run foreman-tasks-delete --state old
+
+    :expectedresults: Old foreman tasks should delete.
+    """
+    result = sat_maintain.cli.Advanced.run_foreman_tasks_delete(options={'state': 'old'})
+    assert 'FAIL' not in result.stdout
+
+
+def test_positive_foreman_taks_delete_planning(sat_maintain):
+    """Delete planning foreman-tasks using advanced procedure run
+
+    :id: 447f5b05-b6a2-4f3a-8e89-8281995ca1cf
+
+    :steps:
+        1. Run satellite-maintain advanced procedure run foreman-tasks-delete --state planning
+
+    :expectedresults: foreman tasks in planning state should delete.
+    """
+    result = sat_maintain.cli.Advanced.run_foreman_tasks_delete(options={'state': 'planning'})
+    assert 'FAIL' not in result.stdout
+
+
+def test_positive_foreman_taks_delete_pending(sat_maintain):
+    """Delete pending foreman-tasks using advanced procedure run
+
+    :id: 6bd3af66-9910-48c4-8cbb-69c3ddd18d6c
+
+    :steps:
+        1. Run satellite-maintain advanced procedure run foreman-tasks-delete --state pending
+
+    :expectedresults: foreman tasks in pending state should delete.
+    """
+    result = sat_maintain.cli.Advanced.run_foreman_tasks_delete(options={'state': 'pending'})
+    assert 'FAIL' not in result.stdout
+
+
+def test_positive_foreman_task_resume(sat_maintain):
+    """Resume paused foreman-tasks using advanced procedure run
+
+    :id: e9afe55b-e3a4-425a-8bfd-a8df6674e516
+
+    :steps:
+        1. Run satellite-maintain advanced procedure run foreman-tasks-resume
+
+    :expectedresults: foreman tasks in paused state should resume.
+    """
+    result = sat_maintain.cli.Advanced.run_foreman_tasks_resume()
+    assert 'FAIL' not in result.stdout
```

Comment:
I think we are not doing the actual scenario testing here. We should try to stop and resume some task, which confirms the correct behavior of this test.

You can improve that later not a blocking comment.

Reply:
yeah, I observed this, lot of tests in this module require setup, which isn't implemented earlier

Reply:
Unresolving the conversation for you to handle in separate